src/messages.rs

```diff
@@ -349,6 +352,50 @@ impl ToField for OutgoingMessages {
     }
 }
 
+fn encode_packet(message: &str) -> String {
+    let data = message.as_bytes();
+
+    let mut packet: Vec<u8> = Vec::with_capacity(data.len() + 4);
+
+    packet.write_u32::<BigEndian>(data.len() as u32).unwrap();
+    packet.write_all(data).unwrap();
+
+    std::str::from_utf8(&packet).unwrap().into()
+}
+
+#[derive(Default, Debug, Clone)]
+pub(crate) struct RequestHandshake {
+    min_version: i32,
+    max_version: i32,
+}
+impl RequestHandshake {
+    pub fn new(min_version: i32, max_version: i32) -> Self {
+        Self { min_version, max_version }
+    }
+    pub fn encode(&self) -> String {
+        format!("v{}..{}", self.min_version, self.max_version)
+    }
+
+    pub fn packet(&self) -> String {
+        // prefix does not get length encoded
+        "API\0".to_owned() + &encode_packet(&self.encode())
```

Comment:
I also conflated these in my implementation, but taking a second look I think these are separate things.
Logically, I think it's better to think of `API\0` as the handshake  and `v100..173` as the first message.
This will simplify the test setup to just one message type.

Reply:
The handshake also requires the absense of the leading null byte. This means we can't use [RequestMethod::encode()](https://github.com/wboayue/rust-ibapi/blob/8f2ab14beee94172e81404ae8cb040afbbac2923/src/messages.rs#L370) method on the handshake.

```txt
b'API\x00\x00\x00\x00\tv100..157'
   
//  RequestMethod { fields: vec!["v176..180"] }::encode()
// raises "Message header length invalid" in TWS / Gateway
b'API\x00\x00\x00\x00\tv100..157\0'
```

So the MockServer would have to:
- have different handling functions for the Message and Handshake (we extend read_prefix()) - both have different reading, encoding handling.
Or
- create a separate struct for the handshake and have a single handling function in the MockServer (we could also simplify the handling in the Connection too)